jupyterhub/tests/test_app.py

```diff
@@ -39,6 +41,24 @@ def test_token_app():
     assert re.match(r'^[a-z0-9]+$', out)
 
 
+def test_raise_error_on_missing_specified_config():
+    """
+    Using the -f or --config flag when starting JupyterHub should require the
+    file to be found and exit if it isn't.
+    """
+    process = Popen(
+        [sys.executable, '-m', 'jupyterhub', '--config', 'not-available.py']
+    )
+    # wait inpatiently for the process to exit
+    for i in range(100):
```

Comment:
Can we use something like https://docs.python.org/3/library/subprocess.html#subprocess.run which will give us the return code of the process without having to poll like this. My guess is that you poll and then kill the process in case it doesn't error out and keeps running. If this guess is right maybe we can add a comment about this so that people from the future also realise this.

Reply:
I didn't find a timeout flag or similar for the process created using run(), so I got stuck there on test-failure scenarios where the process actually just start up.

> Run the command described by args. Wait for command to complete, then return a CompletedProcess instance.

So this was the path I choose in the end, after trying to use .run() first.

Reply:
What do you think of adding a comment here to say exactly what you just said "this looks like it should not be needed (see `subprocess.run()`) but because we want a timeout we do it ourselves"? (In a new Pr as I am merging this one now)

Reply:
Ah yepp it is a very relevant comment since it answers a actual question raised! I created #2826